### DISCUSSION TITLE
using format binary within the archives yaml section repackages the artifacts making them unusable ?

### DISCUSSION THREAD

**Discussion by karlmutch** [goreleaser/goreleaser#3262] [Q&A] [answered]
When performing a snapshot build the binaries produced by a build execute correctly when extracted from the dist directory.

When using the release build the artifacts uploaded to github are repackaged binaries as ar libraries and do run run correctly when downloaded from the github release.

The project is at github.com/karlmutch/duat, using go 1.18.

The goreleaser file contains the following sections, and results in the following release of ar archives rather then pure binaries, https://github.com/karlmutch/duat/releases/tag/0.17.0-rc.5

```
release:
  draft: true
  prerelease: auto
```

```
archives:
- format: binary
```

I have tried various approaches including defining an archive using the ```files: none*``` trick and the gz format but that failed as I have multiple artifacts from the build step which appears not to be able to have an independent archive per artifact.

So my question is how can I perform a publish step that can collect the artifacts and publish without repacking the binaries as ar archives ?

**Comment by caarlos0** [comment, marked as answer]
`format: binary` does nothing to the binaries: https://github.com/goreleaser/goreleaser/blob/31d0e54894d73f5116abee3cdb585faf611ecf6f/internal/pipe/archive/archive.go#L105

if I `goreleaser build` in your repo, the produced binaries do not work as well:

```bash
$ goreleaser build --rm-dist --skip-validate --single-target                                                                                                                                                                                                                                                                                 
# ....

$ file ./dist/*/* | grep -v proxy
./dist/artifact_linux_amd64_v1/artifact:                 current ar archive
./dist/github-release_linux_amd64_v1/github-release:     current ar archive
./dist/license-detector_linux_amd64_v1/license-detector: current ar archive
./dist/semver_linux_amd64_v1/semver:                     current ar archive
./dist/stencil_linux_amd64_v1/stencil:                   current ar archive 
```

a snapshot works, so it seems to me its something in that tag:

```bash
$ goreleaser build --rm-dist --skip-validate --single-target --snapshot                                                                                                                                                                                                                                                                                 
# ....

$ file ./dist/*/* | grep -v proxy
./dist/artifact_linux_amd64_v1/artifact:                 ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=QaKDsOED6qTGG85LYlsQ/-ttu3x7sIZK8vOUhfGtU/L3uTNTrmFl8EJJhVqBF1/oJ0W7WWuvL22KNib9HFZ, not stripped
./dist/github-release_linux_amd64_v1/github-release:     ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=26mdTAayG1xbfXGt5L8S/fc42iZijCBEYj72JElA8/eudH1Lioq_3qkaqFf9wJ/p_JnHP-7Zr4rh1VatPps, not stripped
./dist/license-detector_linux_amd64_v1/license-detector: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=OWB3UJ4i6aNpacCcJCLa/pcuyYsYVuSEooz5YHxcI/QMBOU7HGRWUuLSDgo-wU/CGsJcn4KXNWktRAUjriG, not stripped
./dist/semver_linux_amd64_v1/semver:                     ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=XavAp2RlcJFz1QSzsmRF/GuTH58vjHjvnXu1y3bTH/U3H-U_-qvihVXsXd4H5h/hWPXJbC22RhpvtHIX3KQ, not stripped
./dist/stencil_linux_amd64_v1/stencil:                   ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=g0JcFIWfa6D2lcr0yk7h/JeQo4hozb5MA0eM8dZ4K/VKnuya7chMKo_ypydfqi/70wX7IzHeTlsU37qov3l, not stripped
````

**Reply by caarlos0** [reply]
just realized it might be related to go mod proxying... I bet if you disable it, it'll work (as snapshot does disable it too).

not sure why though

**Reply by caarlos0** [reply]
fixed: https://github.com/karlmutch/duat/pull/139

should use `main` instead of `dir`. Dir is where the `go.mod` file is at. Will clarify docs as well.

**Reply by karlmutch** [reply]
Many thanks for your changes.  I also made an additonal fix which was to make the `main` path have a prefix of `./`
